insights/client/__init__.py

```diff
@@ -255,22 +255,54 @@ def install(self, new_egg):
         """
         returns (dict): {'success': True if the new core was installed successfull else False}
         """
+        import os
+        from shutil import copyfile
+
         success = False
         if not new_egg:
-            logger.debug('Must provide a valid Core installation path.')
-            return {'success': success}
+            the_message = 'Must provide a valid Core installation path.'
+            logger.debug(the_message)
+            return {'success': success, 'message': the_message}
 
         logger.debug("Installing the new Core %s", new_egg)
 
-        from setuptools.command import easy_install
+        # Make sure /var/lib/insights exists
         try:
-            easy_install.main(["-U", new_egg])  # WHY DOESNT THIS RETURN ANYTHING
-            success = True
-        except:
-            logger.debug("There was an issue installing the new Core.")
-            logger.debug("Please ensure a proper path to the Core.")
-            logger.debug("Please ensure you are running as root, or a user with root privileges.")
-
+            if not os.path.isdir(constants.insights_core_lib_dir):
+                logger.debug("Creating directory %s for the Core." % (constants.insights_core_lib_dir))
+                os.mkdir(constants.insights_core_lib_dir)
+        except OSError:
+            message = "There was an error creating %s for Core installation." % (constants.insights_core_lib_dir)
+            raise OSError(message)
+
+        # Copy the NEWEST egg to /var/lib/insights/last_stable.egg
+        old_newest_egg = constants.insights_core_rpm
```

Comment:
This block of code is confusing to me.  Let me just lay out how I think the logic should be and see what you think:

- If there is a `last_stable` and `newest` egg: Copy `newest` to `last_stable`, overwriting the current egg at `last_stable`.
- If there is a `last_stable` but no `newest`: Do nothing (though I think this state will be very rare)
- If there is no `last_stable` but a `newest`: Copy `newest` to `last_stable`
- If there is neither `last_stable` nor `newest`: Do nothing (since there's nothing to do)

So the logic can be reduced to this:  If there's a `newest` egg, copy it to `last_stable`.  I don't think anything should be done with the RPM egg.  It's built in to the fallback mechanism in the client script already.

One question that comes to mind:  How can the script determine if the `newest` egg's last run was actually successful?